packages/data/src/redux-store/index.js

```diff
@@ -176,21 +176,24 @@ export default function createReduxStore( key, options ) {
 			lock( store, privateRegistrationFunctions );
 			const resolversCache = createResolversCache();
 
-			const actions = mapActions(
-				{
-					...metadataActions,
-					...options.actions,
-				},
-				store
-			);
+			function bindAction( action ) {
+				return ( ...args ) =>
+					Promise.resolve( store.dispatch( action( ...args ) ) );
```

Comment:
Does it not turn synchronous results into promises? Or does dispatch always return a promise?

Reply:
Yes, dispatch always returns a promise, for historical reasons. I would love to get rid of that, because it doesn't make much sense -- not all actions are async. But that would break code like `dispatch( foo ).bar().then( ... )`.